src/OpenLoco/src/GameCommands/VehicleOrderInsert.cpp

```diff
@@ -0,0 +1,103 @@
+#include "Entities/EntityManager.h"
+#include "GameCommands/GameCommands.h"
+#include "Objects/VehicleObject.h"
+#include "Ui/WindowManager.h"
+#include "Vehicles/OrderManager.h"
+#include "Vehicles/Orders.h"
+#include "Vehicles/Vehicle.h"
+#include "World/StationManager.h"
+#include <OpenLoco/Interop/Interop.hpp>
+
+using namespace OpenLoco::Interop;
+using namespace OpenLoco::Vehicles;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x0047036E
+    static uint32_t vehicleOrderInsert(const VehicleOrderInsertArgs& args, uint8_t flags)
+    {
+        auto* head = EntityManager::get<VehicleHead>(args.head);
+        if (head == nullptr)
+        {
+            return GameCommands::FAILURE;
+        }
+
+        GameCommands::setPosition(head->position);
+        if (!(flags & GameCommands::Flags::apply))
```

Comment:
i don't quite see why this was moved to here. It makes more sense to leave it at just before the modifying logic on L75